pkg/kube/names.go

```diff
@@ -2,25 +2,35 @@ package kube
 
 import (
 	"bytes"
+	"math"
 	"strings"
 )
 
 // ToValidName converts the given string into a valid Kubernetes resource name
 func ToValidName(name string) string {
-	return toValidName(name, false)
+	return toValidName(name, false, math.MaxInt32)
```

Comment:
Just trying to preserve the previous behavior, in practice perhaps a default limit closer to `math.MaxInt16` or even `math.MaxInt8` would be appropriate?

Reply:
Somewhat related to https://github.com/jenkins-x/jx/issues/2864#issuecomment-457594942.